src/test/lib/PathTest.cpp

```diff
@@ -0,0 +1,106 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    MsQuic Path Unittest
+
+--*/
+
+#include "precomp.h"
+#ifdef QUIC_CLOG
+#include "PathTest.cpp.clog.h"
+#endif
+
+struct PathTestContext {
+    CxPlatEvent ShutdownEvent;
+    MsQuicConnection* Connection {nullptr};
+    bool PeerAddrChanged {false};
+
+    static QUIC_STATUS ConnCallback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+        PathTestContext* Ctx = static_cast<PathTestContext*>(Context);
+        Ctx->Connection = Conn;
+        if (Event->Type == QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE) {
+            Ctx->Connection = nullptr;
+            Ctx->ShutdownEvent.Set();
+        }
+        else if (Event->Type == QUIC_CONNECTION_EVENT_PEER_ADDRESS_CHANGED) {
+            Ctx->PeerAddrChanged = true;
```

Comment:
Have a `CxPlatEvent` here you set, so the main/test thread can just wait, instead of poll.